[2.5.2] Monitoring ignoreNamespaceSelectors

westlywright opened the pull request:
Exposes the `ignoreNamespaceSelectors` field to the prometheus config.

Additionally this sets max-width on our tooltips. I find long tooltips on a wide screen hard to read and super obtrusive. 50vw sets a nice middle ground. This can be backed out though if we'd like to discuss it further. 

rancher/dashboard#1129

vincent99 commented:
Conflicts..

This is a confusing "negative option" and isn't the description backwards?  "Ignore Namespace Selectors ... If true, ... only ... the namespace they are deployed into."  So if ignore namespaces selectors is on, only some namespaces are...selected?

Flip the value and meaning so it's simpler to explain, and if you have to explain both sides of the option it should be a radio instead of a checkbox. Check with someone to confirm exactly what it means but something like

```
Namespace Selectors
( ) Enforced: Monitors can only discover resources in their own namespace.
(•) Ignored: Monitors can discover resources in any namespace.
```

westlywright commented:
Updated to Radio Group and updated the translations as well after some confirmation from @aiyengar2 . Considering the double negative param name we thought it would be better to change the label and wording so that hopefully its a bit clearer.

![Screen Shot 2020-10-09 at 11 45 13 AM](https://user-images.githubusercontent.com/858614/95620364-ea3e8b80-0a24-11eb-9ee3-8ebd6354a05e.png)

aiyengar2 commented on assets/translations/en-us.yaml:
```diff
@@ -782,6 +784,12 @@ monitoring:
+      ignoreNamespaceSelectors:
+        label: Use Namespace Selectors
+        radio:
+          enforced: "Use: Monitors can access resources in the namespace specified in the namespaceSelector field"
```
```suggestion
          enforced: "Use: ServiceMonitors & PodMonitors can be configured to monitor resources across namespaces using the `namespaceSelector` field"
```

Not sure if the \` works as expected, but would be nice if we can highlight `namespaceSelector` to make it clear that it is a field of the ServiceMonitor / PodMonitor spec

aiyengar2 commented on assets/translations/en-us.yaml:
```diff
@@ -782,6 +784,12 @@ monitoring:
+        label: Use Namespace Selectors
+        radio:
+          enforced: "Use: Monitors can access resources in the namespace specified in the namespaceSelector field"
+          ignored: "Ignore: Monitors can only access resources in the deployed namespace that they are deployed in"
```
```suggestion
          ignored: "Ignore: ServiceMonitors & PodMonitors can only monitor resources in the namespaces they are deployed within. The `namespaceSelector` field will be ignored."
```
Not sure if the \` works as expected, but would be nice if we can highlight `namespaceSelector` to make it clear that it is a field of the ServiceMonitor / PodMonitor spec

aiyengar2 commented on assets/translations/en-us.yaml:
```diff
@@ -782,6 +784,12 @@ monitoring:
+        radio:
+          enforced: "Use: Monitors can access resources in the namespace specified in the namespaceSelector field"
+          ignored: "Ignore: Monitors can only access resources in the deployed namespace that they are deployed in"
+        help: Ignoring Namespace Selectors allows Cluster Admins to limit teams from monitoring resources outside of namespaces they have permissions to but can break the functionality of Apps that rely on setting up Monitors that scrape targets across multiple namespaces, such as Istio.
```
Can this be a warning instead?

```
Ignoring `namespaceSelectors` can break the functionality of Apps that try to monitor targets across namespaces using a single ServiceMonitor / PodMonitor, such as Istio.
```
